specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -472,71 +417,75 @@ namespace Microsoft.Windows.PushNotifications
 
         // More detailed error code in addition to the ChannelStatus state.
         HRESULT ExtendedError{ get; };
 
-        // The Status of the ChannelComplete operation
+        // The Status of the CreateChannel operation
         PushNotificationChannelStatus Status { get; };
     };
 
-    static runtimeclass PushNotificationManager
+    [feature(Feature_PushNotifications)]
+    runtimeclass PushNotificationManager
     {
-        // Register an activator using an ActivationInfo context and cache the flag for unregister
-        static void RegisterActivator(PushNotificationActivationInfo details);
+        // Gets a Default instance of a PushNotificationManager
+        static PushNotificationManager Default{ get; };
+
+        // Registers an application for Notifications
+        // For Unpackaged applications, registers support for protocol activations
+        // For packaged applications with Background Infrastructure support, registers Push Trigger with the Background Infrastructure
+        // For packaged applications without Background Infrastructure support, registers a COM Service
+        void Register();
 
-        // Unregister any activator given PushNotificationRegistrationActivators
-        // 1) If kind = PushTrigger is specified, the trigger itself will be removed
-        // 2) If kind = ComActivator is specified, the Project Reunion Background Task component will no longer act as an InProc COM Server
-        // 3) If kind = ProtocolActivator is specified, the application will be unregistered from the long running process that handles activation
-        static void UnregisterActivator(PushNotificationRegistrationActivators activators);
+        // Packaged Apps: Force COM to launch a new process
+        // Unpackaged apps: Remove the Foreground Sink to force Long Running Process Singleton to launch a new process instance.
+        void Unregister();
 
-        // Unregister all activators registered for the application
-        static void UnregisterAllActivators();
+        // Disables Push completely for Packaged and Unpackaged apps. Register() needs to be called again for Push to work
+        void UnregisterAll();
 
         // Request a Push Channel with an encoded RemoteId from WNS. RemoteId is an AAD identifier GUID
-        static Windows.Foundation.IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> CreateChannelAsync(Guid remoteId);
+        Windows.Foundation.IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> CreateChannelAsync(Guid remoteId);
 
-        // Applications will call this to check which flags are supported for RegisterActivator
-        static Boolean IsActivatorSupported(PushNotificationRegistrationActivators activators);
+        // Event handler for Push Notifications
+        // NOTE: You must register your foreground handlers before calling Register() or else it will throw.
```

Comment:
Is it really only "foreground" handlers here? Background handlers do not have this requirement?

Reply:
GetActivatedEventArgs is the equivalent of background handlers. Handling background activation is a one-time event and to account for synchronization issues we are forcing developers to register handlers first.

Reply:
This uses the same pattern as AppNotificationManager and should be easy for devs to comprehend since it follows similar patterns.